rasa/components/providers/training_tracker_provider.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+from typing import Dict, Text, Any, List
+
+from rasa.engine.graph import GraphComponent, ExecutionContext
+from rasa.engine.storage.resource import Resource
+from rasa.engine.storage.storage import ModelStorage
+from rasa.shared.core.domain import Domain
+from rasa.shared.core.generator import TrackerWithCachedStates, TrainingDataGenerator
+from rasa.shared.core.training_data.structures import StoryGraph
+
+
+class TrainingTrackerProvider(GraphComponent):
+    """Provides training trackers to policies based on training stories."""
+
+    def __init__(self, config: Dict[Text, Any]) -> None:
+        self._config = config
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        return {
+            "remove_duplicates": True,
```

Comment:
I think these parameters should (only) be defined in `TrainingDataGenerator` and here we should just pass them on (via some `TrainingDataGenerator.get_default_config()`

And why don't we make `TrainingDataGenerator` a GraphComponent itself?  `generate_trackers` wouldn't need the instantiation of a `TrainingDataGenerator` then (and it doesn't need to be instantiated multiple times because configs don't change / cannot be specified in `generate_trackers` anyway.

Reply:
That's a good idea 👍🏻 Will do so 👍🏻

Reply:
@ka-bu Actually I now remember why I didn't do this. Doing this would mean to create a dependency to `rasa.engine` in the `rasa.shared` package which in turn would mean that we'd need to move the entire engine package to `rasa.shared` (as `rasa.shared` is not allowed to import anything outside of `rasa.shared` which I wanted to avoid. 

> I think these parameters should (only) be defined in TrainingDataGenerator

It's somewhat fine as we have the unit test to ensure these params are correct, right?

Reply:
Ah ok, so we need the wrapper here. But I'd still move the parameters/tests. Because otherwise we're only half-way respecting the engine/shared border (and the wrapper just begin a wrapper), don't we? Because those parameters are internals of `TrainingDataGenerator` and if we change the internals there, we'd have to adapt the parameters here *and* the unit tests here as well. 
(And if we want tests here, we could mock.patch the TrainingDataGenerator in the tests then and just ensuring that creation and generate are called - cause that's all that this wrapper should do / we know "here")

Reply:
I think the tests are valid as they test whether we are actually wrapping the `TrainingDataGenerator` correctly. if we'd mock the tracker generator then we can't catch potentially breaking changes (such as if somebody renames the params). 

How would you move the params over due `TrainingDataGenerator`? I fail to see an elegant way to do so besides inspecting the method signatures which in my opinion goes to far.

Reply:
My workaround would be to adapt `TrainingDataGenerator` so that it looks like a GraphComponent in some ways but we don't call it like that, i.e. has a `config` ,  instead of `self.<param>` do `self.config[<param>]` , adding properties to account for the missing `self.domain` for example ,.... and yeah that is a lot of changes. ... Agree that without that we should keep the tests. Then let's just keep it as is (?).

Reply:
I gave it another thought and decided to keep it as it is for now. I think you've very good points but I don't think it's the best use of our resources for the moment. Also I'd like to keep the changes to existing things minimal so we have less to look through in case some of the model regression tests or something else is giving weird results